### PR TITLE
fix: Fix the abnormal problems that occur when the license is unbound

### PR DESCRIPTION
**Pull request by ssongliu**

### CODE REVIEW

File: frontend/src/api/index.ts [hunk 1]
```diff
@@ -64,6 +64,11 @@ class RequestHttp {
...
                     return Promise.reject(data);
                 }
+                if (data.code == ResultEnum.NodeUnBind) {
+                    globalStore.currentNode = 'local';
+                    window.location.reload();
+                    return;
+                }
                 if (data.code == ResultEnum.ERRGLOBALLOADDING) {
                     globalStore.setGlobalLoading(true);
                     globalStore.setLoadingText(data.message);
```
Comment (wanghe-fit2cloud) [inline frontend/src/api/index.ts:74]: The provided code snippet modifies the `RequestHttp` class to handle specific cases related to HTTP requests. Here are the observations and a few suggestions:

### Observations:
1. **Reload on Unbound Node**:
   - If the response `data.code` is equal to `ResultEnum.NodeUnbind`, it updates the store `globalStore.currentNode` with `'local'` and reloads the page. This seems logical but ensure that this logic aligns with your application's requirements.

2. **Error Handling**:
   - The code also includes a condition where if `data.code` equals `ResultEnum.ERRGLOBALLOADDING`, it sets the loading state in the `globalStore`. However, the actual handling of displaying or logging error messages would depend on additional context.

3. **Return Value**:
   - When `data.code` is `ResultEnum.CodeSuccess`, the function returns the parsed data as promised.
   - In other cases, whether it's an unbound node, an error loading globally, or any other failure, the function handles them by reloading the page or setting the loading status but does not immediately return anything else.

### Suggestions:
1. **Consistent Loading Behavior**: Consider ensuring consistent behavior when dealing with different types of failures. For example, you might want to log the error message instead of just showing a reload.

2. **Code DRY Principle**: Look for opportunities to dry up repeated code or patterns across similar handlers. Currently, all paths lead to a reload, which could be done more generically.

3. **Error Messages**: Implement proper feedback mechanisms (e.g., popups or toast notifications) for users about errors such as network issues or permission problems beyond merely reloading the page.

4. **Type Safety**: Ensure type safety is maintained throughout the codebase. You can use TypeScript or static typing features to catch potential runtime errors early.

5. **Scalability**: As applications grow complex, consider using pattern-matching or decorators to streamline these conditions rather than repetitive checks.

6. **Testing**: Test various scenarios, including edge cases, to ensure robustness. This will help identify any unforeseen behaviors or bugs.

Here's a modified version incorporating some of these suggestions:

```javascript
class RequestHttp {
    // Other methods...

    handleError(data) {
        if ([ResultEnum.ErrGlobalLoading].includes(data.code)) {
            globalStore.setGlobalLoading(true);
            globalStore.setLoadingText(data.message);
            alert('Error occurred: ' + data.message); // Add proper notification
        } else if (data.code === ResultEnum.NodeUnbind) {
            globalStore.currentNode = 'local';
            // Log something before redirecting
            console.log('Redirected due to unbound node.');
            window.location.reload();
        } else {
            AlertManager.alert(data.code, 'System Error'); // Using hypothetical AlertManager utility
        }

        return null; // Returning undefined or empty object based on project needs
    }

    sendHttpRequest(requestParams) {
        return fetch(this.baseUrl + requestParams.url)
            .then(response => response.json())
            .then(jsonData => this.handleResponse(jsonData))
            .catch(error => this.handleError({ code: error.status }));
    }
}
```

This approach improves code readability and maintainability while adding basic error handling.

File: frontend/src/enums/http-enum.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ export enum ResultEnum {
     ERRAUTH = 406,
     ERRGLOBALLOADDING = 407,
     ERRXPACK = 410,
+    NodeUnBind = 411,
     TIMEOUT = 20000,
     TYPE = 'success',
 }
```
Comment (wanghe-fit2cloud) [inline frontend/src/enums/http-enum.ts:18]: The provided code snippet is syntactically correct and does not contain any immediate irregularities or potential issues. However, it seems like you intended to add a new `ResultEnum` constant named `NodeUnBind`. The addition of this new value looks appropriate and should work correctly within the context of your existing enum.

If there are specific areas where this code might be used, such as in error handling mechanisms, testing cases, or application logic that deals with different HTTP status codes, additional checks may be necessary. Otherwise, this change can be integrated smoothly into your system without affecting its functionality.

For optimization suggestions:
1. **Documentation**: Ensure that if you document these constants somewhere else (like in comments), they are also included here for clarity.
2. **Scalability**: If future developments involve expanding error enums, consider whether adding more values will lead to unnecessary complexity.
3. **Codebase Consistency**: Make sure all users of this enum understand its purpose and usage patterns.

Overall, the current version of the code is valid and ready for integration.

**Review by wanghe-fit2cloud — COMMENTED** [review]
/lgtm
